tests/solidity/test_open_short.py

```diff
@@ -0,0 +1,205 @@
+        )
+        self.assertEqual(  # share price
+            self.hyperdrive.market_state.share_price,
+            market_state_before.share_price,
+            msg=f"{self.hyperdrive.market_state.share_price=} is not correct",
+        )
+        self.assertEqual(  # longs outstanding
+            self.hyperdrive.market_state.longs_outstanding,
+            market_state_before.longs_outstanding,
+            msg=f"{self.hyperdrive.market_state.longs_outstanding=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.long_average_maturity_time,
+            0,
+            msg=f"{self.hyperdrive.market_state.long_average_maturity_time=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.long_base_volume,
+            0,
+            msg=f"{self.hyperdrive.market_state.long_base_volume=} is not correct",
+        )
+        # TODO: once we add checkpointing we will need to switch to this
+        # self.hyperdrive.market_state.long_base_volume_checkpoints(checkpoint_time),
+        # checkpoint_time = maturity_time - self.position_duration
+        self.assertEqual(
+            self.hyperdrive.market_state.long_base_volume,
+            0,
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.shorts_outstanding,
+            market_state_before.shorts_outstanding + unsigned_bond_amount,
+            msg=f"{self.hyperdrive.market_state.shorts_outstanding=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.short_average_maturity_time,
+            maturity_time,
+            msg=f"{self.hyperdrive.market_state.short_average_maturity_time=} is not correct",
+        )
+        self.assertEqual(
+            self.hyperdrive.market_state.short_base_volume,
+            abs(base_amount),
+            msg=f"{self.hyperdrive.market_state.short_base_volume=} is not correct",
+        )
+        # TODO: once we add checkpointing we will need to switch to this
+        # self.hyperdrive.market_state.short_base_volume_checkpoints(checkpoint_time),
+        self.assertEqual(
+            self.hyperdrive.market_state.short_base_volume,
+            abs(base_amount),
+            msg=f"{self.hyperdrive.market_state.short_base_volume=} is not correct",
+        )
+
+    def test_open_short_failure_zero_amount(self):
+        """shorting bonds with zero base fails"""
+        with self.assertRaises(AssertionError):
+            self.hyperdrive.open_short(self.bob.wallet, 0)
+
+    def test_open_short_failure_extreme_amount(self):
+        """shorting more bonds than there is base in the market fails"""
+        # TODO: Shouldn't this be a function of the contribution amount?
+        # The max amount of base does not equal the amount of bonds, it is the result of base_pm.get_max_long
+        bond_amount = self.hyperdrive.market_state.bond_reserves * 2
```

Comment:
either way you get an amount that's too big, but shorts take in bonds as the amount 🤷

Reply:
look at get_max_short to investigate what should be passed in, which turned into https://github.com/element-fi/elf-simulations/pull/288
using one of the reserves as a proxy for the max short is a big guess, but I guess base_reserves is a better estimate, since it's bigger
will switch to using that one like in solidity